src/utils/formatting.rs

```diff
@@ -86,3 +86,81 @@ pub fn nice_directory_display(path: &Path) -> Cow<str> {
         to_utf(path)
     }
 }
+
+/// Struct useful to printing bytes as kB, MB, GB, etc.
+pub struct Bytes(f64);
+
+impl Bytes {
+    const UNIT_PREFIXES: [&'static str; 6] = ["", "ki", "Mi", "Gi", "Ti", "Pi"];
+
+    /// Create a new Bytes.
+    pub fn new(bytes: u64) -> Self {
+        Self(bytes as f64)
+    }
+}
+
+impl std::fmt::Display for Bytes {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let &Self(num) = self;
+
+        debug_assert!(num >= 0.0);
+        if num < 1_f64 {
+            return write!(f, "{} B", num);
+        }
+
+        let delimiter = 1000_f64;
+        let exponent = cmp::min((num.ln() / 6.90775).floor() as i32, 4);
```

Comment:
The constant `6.90775` here is arcane-event-horizon-magic, but it works.

If someone wants to refactor and it ends up as fast as it is, that's great, but I'm not the one who's gonna touch this relic.

I actually wanted MB instead of MiB, but I don't know how to the change the code to do so xD.